cypress/integration/purchase/purchase_order_complete_change_complete.js

```diff
@@ -0,0 +1,170 @@
+import { BPartner } from '../../support/utils/bpartner';
+import { BPartnerLocation } from '../../support/utils/bpartner_ui';
+import { DiscountSchema } from '../../support/utils/discountschema';
+import { ProductCategory } from '../../support/utils/product';
+import { PackingMaterial } from '../../support/utils/packing_material';
+import { PackingInstructions } from '../../support/utils/packing_instructions';
+import { PackingInstructionsVersion } from '../../support/utils/packing_instructions_version';
+import { purchaseOrders } from '../../page_objects/purchase_orders';
+import { Builder } from '../../support/utils/builder';
+import { getLanguageSpecific, humanReadableNow } from '../../support/utils/utils';
+import { PurchaseOrderStatus } from '../../support/utils/constants';
+
+describe('Create Purchase order - material receipt - invoice', function() {
+  const date = humanReadableNow();
+  const productForPackingMaterial = `ProductPackingMaterial ${date}`;
+  const productPMValue = `purchase_order_testPM ${date}`;
+  const packingMaterialName = `ProductPackingMaterial ${date}`;
+  const packingInstructionsName = `ProductPackingInstrutions ${date}`;
+  const productName1 = `ProductTest ${date}`;
+  const productValue1 = `purchase_order_test ${date}`;
+  const productCategoryName = `ProductCategoryName ${date}`;
+  const productCategoryValue = `ProductCategoryValue ${date}`;
+  const discountSchemaName = `DiscountSchemaTest ${date}`;
+  const priceSystemName = `PriceSystem ${date}`;
+  const priceListName = `PriceList ${date}`;
+  const priceListVersionName = `PriceListVersion ${date}`;
+  const productType = 'Item';
+  const vendorName = `Vendor ${date}`;
+
+  before(function() {
+    Builder.createBasicPriceEntities(priceSystemName, priceListVersionName, priceListName, false);
```

Comment:
All this code appears in the runner to be part of the first test. That makes it harded to discern where the setup stops and the actual tests start.
I suggest to have one or more dedicated "setup" `it`'s to one can see more easily what is going on.

Reply:
Should I put the code from before function in separate * it's * ?

Reply:
imho yes

Reply:
ok